### PR TITLE
feat: use pnpm instead of npm

### PR DESCRIPTION
**Pull request by yomybaby**
This PR resolves several issues related to the Node Package Manager (NPM). PNPM has a good performance:

- Consistent installation: Running `npm i` in some environments results in different outcomes.
- Installation speed
- More explicit module management: PNPM warns about missing packages in package.json.

In this PR, we are using PNPM in Makefile and GitHub Actions.

How to test:
- run dev servers using `pnpm` and check web pages
- `make` build and check the app

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#2515** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2515?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .gitignore
Comment (Yaminyam) [inline .gitignore:1]: It would be a good idea to also add `pnpm-debug.log`.

**Review by lizable — CHANGES_REQUESTED** [review]
I got error "Cannot find module node:events" while building desktop app.

**Review by lizable — COMMENTED** [review]
Tested whether feature works properly using pnpm and it works fine.
Also building desktop app using `Makefile` is also works without error.
It would be good to go after applying @Yaminyam 's suggestion, though.

File: .gitignore
Comment (yomybaby) [reply .gitignore:1]: Done.
```
pnpm-debug.log*
pnpm-error.log*
```

**Review by lizable — APPROVED** [review]
LGTM.

**Review by Yaminyam — COMMENTED** [review]
It would be a good idea to add the contents added to `.gitignore` to `.dockerignore` as well.

**Pull request merged**